File: cs50/sentimental-readability/readability.py
from cs50 import get_string
t = get_string("Text: ")
wornum = 1
sennum = 0
letnum = 0
for i in range(len(t)):
    if t[i].isalpha():
        letnum += 1
    elif t[i].isspace():
        wornum += 1
    elif t[i] == '.' or t[i] == '!' or t[i] == '?':
        sennum += 1
l = letnum / wornum * 100
s = sennum / wornum * 100
i = 0.0588 * l - 0.29 * s - 15.8
i = round(i)
if i > 16:
    print("Grade 16+")
elif i < 17 and i > 1:
    print(f"Grade {i}")
else:
    print("Before Grade 1")

File: cs50/readability/readability.c
#include <string.h>
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int sentencecount(string text);
int wordcount(string text);
int lettercount(string text);
int main(void)
{
    string text;
    int letters, words, sentences, index;
    double L, S;
    text = get_string("Text:");
    letters = lettercount(text);
    words = wordcount(text);
    sentences = sentencecount(text);
    L = letters/(float)words * 100;
    S = sentences/(float)words * 100;
    index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index > 16)
        printf("Grade 16+\n");
    else if (index < 1)
        printf("Before Grade 1\n");
    else
        printf("Grade %i\n", index);
}
int lettercount(string text)
{
    int len, count = 0;
    len = strlen(text);
    for (int i = 0; i< len; i++)
    {
        if (isalpha(text[i]) != 0)
            count ++;
    }
    return count;
}
int wordcount(string text)
{
    int len, count = 0;
    len = strlen(text);
    for (int i = 0; i< len; i++)
    {
        if (text[i] == ' ')
            count ++;
    }
    count ++;
    return count;
}
int sentencecount(string text)
{
    int len, count = 0;
    len = strlen(text);
    for (int i = 0; i< len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            count ++;
    }
    return count;
}
